lambdas/tests/e2e/contact-retrieve-transcript.test.ts

```diff
@@ -0,0 +1,117 @@
+import { jest } from '@jest/globals';
+import { S3, SQS } from 'aws-sdk';
+import { generateMockMessageBody } from '../../src/contact-retrieve-transcript/tests/generateMockMessageBody';
+import { getStackOutput } from '../parseCdkOutput';
+import { sendMessage } from '../sendMessage';
+
+/**
+ * TODO: This is a super dirty proof of concept for e2e tests.
+ * It needs cleanup.
+ */
+
+jest.setTimeout(60000);
+
+const localstackEndpoint = 'http://localhost:4566';
+
+const completeOutput: any = getStackOutput('contact-complete');
+const { queueUrl } = completeOutput;
+
+// TODO: modularize all of this setup for reuse
+const s3 = new S3({
+  region: 'us-east-1',
+  endpoint: localstackEndpoint,
+  s3ForcePathStyle: true,
+});
+
+const sqs = new SQS({
+  region: 'eu-west-1',
+});
+
+const lambdaName = 'contact-retrieve-transcript';
+
+export const waitForS3Object = async ({
+  message,
+  retryCount = 0,
+}: {
+  message: ReturnType<typeof generateMockMessageBody>;
+  retryCount?: number;
+}): Promise<S3.GetObjectOutput | undefined> => {
+  const params = {
+    Bucket: 'contact-docs-bucket',
+    Key: message.filePath,
+  };
+
+  let result;
+  try {
+    result = await s3.getObject(params).promise();
+  } catch (err) {
+    if (retryCount < 15) {
+      await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
Similar to below, I'd hope a local s3 would be pretty snappy so we can afford a more aggressive retry, just to try keeping the tests as snappy as possible